### ISSUE TITLE
Caching conda environment in CI

### ISSUE THREAD

**Issue by wkrasnicki** [dask/dask#10020]
Hello

After submitting my PR recently, I've noticed that each test job took about 6 minutes to set up the conda environment. I've noticed that the workflow has no caching mechanism, even though a simple example solution can be found in the official [docs](https://github.com/conda-incubator/setup-miniconda#caching-environments) of the `conda-incubator` action.

What is more, a similar issue has been raised in the [dask distributed](https://github.com/dask/distributed/issues/6853) repo. I have some experience with GitHub Actions and would be happy to submit a similar PR.

**Comment by j-bennet** [comment]
@wkrasnicki Thank you, I see this is already implemented in `dask/distributed`, would be good to have here as well.

**Comment by jrbourbeau** [comment]
Thanks for the issue @wkrasnicki. I personally don't feel much pain around conda environment setup at the moment. Though as long as the caching code wasn't too complex, I'd welcome something that shaved 6 minutes off CI times. @hendrikmakait @crusaderky @fjetter what's your experience been like with cached environments over in `distributed`?

**Comment by hendrikmakait** [comment]
TL;DR: Cache invalidation is hard. Spread the word about caching (and invalidation) on CI and document it in a place where it's easy to discover (likely https://docs.dask.org/en/latest/develop.html#dask-ci-infrastructure), then things will be fine.

From my experience, caching works great most of the time. It's just that when it causes problems, it gets kind of cryptic if you don't know what's happening. I had an issue with this when we merged the PRs for P2P rechunk where changes merged into `dask/dask` would not appear on `dask/distributed` CI due to caching. It took me a while to realize that caching was the culprit and how to invalidate the caches. Documenting such a scenario as a common gotcha and how to resolve this would be beneficial.

**Comment by hendrikmakait** [comment]
https://github.com/dask/distributed/issues/7628 is an example of an issue caused by `dask/dask` and `dask/distributed` getting out of sync due to caching.
